From e7f3e65a4163de14b9316cb874a83fb05ac767f2 Mon Sep 17 00:00:00 2001
From: ksco <numbksco@gmail.com>
Date: Mon, 7 Nov 2022 18:12:41 +0800
Subject: [PATCH] RV32 Modify addi to generate test cases

---
 riscv/insns/addi.h | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/riscv/insns/addi.h b/riscv/insns/addi.h
index 1bb5dced..730ec8db 100644
--- a/riscv/insns/addi.h
+++ b/riscv/insns/addi.h
@@ -1 +1,16 @@
+static int ncase = 2;
+
+if (insn.rd() == 0 && insn.i_imm() != 0) {
+    for (int reg = insn.rs1(); reg < insn.i_imm(); reg++) {
+        for (int i = 0; i < P.VU.VLEN / 32; i++) {
+            printf(
+                "  TEST_CASE(%d, t0, 0x%x, lw t0, 0(a0); addi a0, a0, 4)\n",
+                ncase++,
+                P.VU.elt<type_sew_t<32>::type>(reg, i, false));
+        }
+    }
+
+    printf("---\n");
+}
+
 WRITE_RD(sext_xlen(RS1 + insn.i_imm()));
-- 
2.30.1 (Apple Git-130)

